<?xml version="1.0"?>
<launch>
  <group ns="rtabmap">

    <!-- Odometry -->
    <node pkg="rtabmap_ros" type="rgbd_odometry" name="rgbd_odometry">
      <remap from="rgb/image"       to="/camera/rgb/image_raw"/>
      <remap from="depth/image"     to="/camera/depth/image_raw"/>
      <remap from="rgb/camera_info" to="/camera/depth/camera_info"/>

	  <param name="odom_frame_id" type="string" value="odom"/>
      <param name="frame_id" type="string" value="base_link"/>
      <param name="publish_tf" type="bool" value="true"/>
    </node>

    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap"  args="--delete_db_on_start">
          <param name="frame_id" type="string" value="base_link"/>
          <param name="publish_tf" type="bool" value="false"/>
          <param name="map_frame_id" type="string" value="map"/>
          <param name="odom_frame_id" type="string" value="odom"/>
		  <param name="subscribe_depth" type="string" value="true"/>

          <remap from="rgb/image" to="/camera/rgb/image_raw"/>
          <remap from="depth/image" to="/camera/depth/image_raw"/>
          <remap from="rgb/camera_info" to="/camera/rgb/camera_info"/>

          <param name="queue_size" type="int" value="1"/>

          <!-- RTAB-Map's parameters -->
          <param name="RGBD/AngularUpdate" type="string" value="0.01"/>
          <param name="RGBD/LinearUpdate" type="string" value="0.01"/>
		  <param name="RGBD/OptimizeFromGraphEnd" type="string" value="true"/>
          <param name="Rtabmap/TimeThr" type="string" value="700"/>
          <param name="Mem/RehearsalSimilarity" type="string" value="0.45"/>
    </node>
    
<!--
      <node name="rtabmapviz" pkg="rtabmap_ros" type="rtabmapviz" 
                args="-d $(find rtabmap_ros)/launch/config/rgbd_gui.ini" output="screen">
      <param name="subscribe_scan"              type="bool"     value="false"/>
      <param name="frame_id"                    type="string"   value="base_link"/>
      <param name="wait_for_transform_duration" type="double"   value="0.2"/>
      <remap from="rgb/image"       to="/camera/rgb/image_raw"/>
      <remap from="depth/image"     to="/camera/depth/image_raw"/>
      <remap from="rgb/camera_info" to="/camera/rgb/camera_info"/>
    </node>    
-->

  </group>
</launch>
